roles/pulp_api/tasks/generate_database_fields_key.yml

```diff
@@ -40,7 +41,7 @@
 - name: Ensure correct permissions on new Pulp Database Encryption Key
   file:
     path: '{{ __pulp_db_fields_key_path }}'
-    owner: "{{ pulp_user }}"
+    owner: 'root'
     group: "{{ pulp_group }}"
-    mode: 0600
+    mode: 0640
```

Comment:
It lives in `pulp_certs_dir`, which is owned by pulp.

Wouldn't it make more sense to leave it as `{{ pulp_user }}` and change permissions to 440 ?

Reply:
My understanding is that `640` and `root:$group` is fairly standard for private keys.

Thinking about it more, I think `440` and `root:$group` would be even better, to prevent someone from accidentally writing changes to the file during a troubleshooting session (`root` can do what it wants, but at least vim will force you to use `:wq!` instead of `:wq` when writing to a read-only file as `root`)

Losing the key would mean losing all encrypted fields in the DB, so it seems imperative to prevent that from happening by mistake.

It does have the side effect that someone who really needed to modify the key would require root access to do so. Given how destructive accidental key modification would be, that seems OK?

I think with `440` and `$user:$group` you still face the possibility that the service could modify the file. (If $user or the application were compromised and it's owned by $user, an attacker could chmod and modify or delete the key, causing irrecoverable data loss)

All that said, let me know how you want to proceed in this area and I'll make whichever change you prefer.

Reply:
Sorry for the delay.

At this point, I think it is configured well enough. `640` and `root:$group` works, and I'd very much like to merge this PR.

Very good point though!